### PR TITLE
Add tests for WriteProtocolMessage and ReadProtocolMessage

### PR DESCRIPTION
**Pull request by eliben**

### CODE REVIEW

File: io_test.go [hunk 1]
```diff
@@ -287,3 +287,51 @@ func TestReadWriteWithCodec(t *testing.T) {
...
+
+	buf := new(bytes.Buffer)
+	err := WriteProtocolMessage(buf, req)
+	if err != nil {
+		t.Error(err)
+	}
+
+	want := "Content-Length: 75\r\n\r\n{\"seq\":25,\"type\":\"request\",\"command\":\"cancel\",\"arguments\":{\"requestId\":24}}"
+	if buf.String() != want {
+		t.Errorf("got %v, want %v", buf.String(), want)
```
Comment (polinasok) [inline io_test.go]: %#v

File: io_test.go [hunk 2]
```diff
@@ -287,3 +287,51 @@ func TestReadWriteWithCodec(t *testing.T) {
...
+	}
+}
+
+func TestReadProtocolMessage(t *testing.T) {
+	input := "Content-Length: 75\r\n\r\n{\"seq\":25,\"type\":\"request\",\"command\":\"cancel\",\"arguments\":{\"requestId\":24}}"
```
Comment (polinasok) [inline io_test.go]: Same as above, so use a global?

File: io_test.go (hunk 1 above)
Comment (polinasok) [inline io_test.go]: %#v

File: io_test.go [hunk 3]
```diff
@@ -287,3 +287,51 @@ func TestReadWriteWithCodec(t *testing.T) {
...
+	reader := bufio.NewReader(strings.NewReader(input))
+
+	msg, err := ReadProtocolMessage(reader)
+	if err != nil {
+		t.Error(err)
+	}
+
+	wantreq := &CancelRequest{
```
Comment (polinasok) [inline io_test.go]: Same comment as above.
Another option is to test Read and Write in the same test function, so you can compare that writing and reading the same thing brings you back to what you started with.

File: io_test.go (hunk 1 above)
Comment (eliben) [reply io_test.go]: Done

Comment (eliben) [reply io_test.go]: Done

File: io_test.go (hunk 2 above)
Comment (eliben) [reply io_test.go]: Done

File: io_test.go (hunk 3 above)
Comment (eliben) [reply io_test.go]: Done

File: io_test.go [hunk 4]
```diff
@@ -288,50 +288,40 @@ func TestReadWriteWithCodec(t *testing.T) {
...
 
-func TestWriteProtocolMessage(t *testing.T) {
-	req := &CancelRequest{
-		Request: Request{
-			ProtocolMessage: ProtocolMessage{
-				Type: "request",
-				Seq:  25,
-			},
-			Command: "cancel",
+const cancelMsgString = "Content-Length: 75\r\n\r\n{\"seq\":25,\"type\":\"request\",\"command\":\"cancel\",\"arguments\":{\"requestId\":24}}"
```
Comment (polinasok) [inline io_test.go]: nit: Msg => Req or Request (since you can also have response msg)

Comment (eliben) [reply io_test.go]: Done

**Pull request merged**